recipes/icassp_2023/baseline/evaluate.py

```diff
@@ -0,0 +1,202 @@
+                f"The combined score is {si} (haspi {si_haspi}, hasqi {si_hasqi})"
+            )
+            csv_lines.append([scene, listener, str(si), str(si_haspi), str(si_hasqi)])
+
+            if cfg.evaluate.cal_unprocessed_si:
+                if cfg.evaluate.set_random_seed:
+                    scene_md5 = int(
+                        hashlib.md5(scene.encode("utf-8")).hexdigest(), 16
+                    ) % (10**8)
+                    np.random.seed(scene_md5)
+
+                fs_unproc, unproc = wavfile.read(
+                    os.path.join(cfg.path.scenes_folder, f"{scene}_mix_CH1.wav")
+                )
+                unproc = unproc / 32768.0
+                si_unproc_haspi = haspi_v2_be(
+                    xl=ref[:, 0],
+                    xr=ref[:, 1],
+                    yl=unproc[:, 0],
+                    yr=unproc[:, 1],
+                    fs_signal=fs_ref_anechoic,
+                    audiogram_l=audiogram_left,
+                    audiogram_r=audiogram_right,
+                    audiogram_cfs=cfs,
+                )
+                si_unproc_hasqi = hasqi_v2_be(
+                    xl=ref[:, 0],
+                    xr=ref[:, 1],
+                    yl=unproc[:, 0],
+                    yr=unproc[:, 1],
+                    fs_signal=fs_ref_anechoic,
+                    audiogram_l=audiogram_left,
+                    audiogram_r=audiogram_right,
+                    audiogram_cfs=cfs,
+                )
+
+                si_unproc = np.mean([si_unproc_haspi, si_unproc_hasqi])
+
+                logger.info(
+                    f"The unprocessed signal combined evaluation score is {si_unproc}"
+                )
+                unproc_csv_lines.append(
+                    [
+                        scene,
+                        listener,
+                        str(si_unproc),
+                        str(si_unproc_haspi),
+                        str(si_unproc_hasqi),
+                    ]
+                )
+
+    with open(si_file, "w") as csv_f:
+        csv_writer = csv.writer(
+            csv_f, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+        )
+        for line in csv_lines:
+            csv_writer.writerow(line)
+    score_dict = read_csv_scores(si_file)
+    ave_score = np.mean(list(score_dict.values()))
+    logger.info(
+        "si.csv exists, and the average combined score is {:.4f}".format(ave_score)
+    )
+
+    if cfg.evaluate.cal_unprocessed_si:
+        with open(unproc_si_file, "w") as csv_f:
+            csv_writer = csv.writer(
+                csv_f, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            for line in unproc_csv_lines:
+                csv_writer.writerow(line)
+        score_dict = read_csv_scores(unproc_si_file)
+        ave_score = np.mean(list(score_dict.values()))
+        logger.info(
+            "si_unproc.csv exists, and the average HASPI scores for unprocessed scenes is {:.4f}".format(
```

Comment:
This could be an f-string as in other logging statements, would take the form...

```suggestion
            f"si_unproc.csv exists, and the average HASPI scores for unprocessed scenes is {mean_score:.4f}"
```

(accounting for above suggestion(s) on variable name.

Reply:
f-strings are more clear than `.format()`, especially when you have several variables in the same string.